lnwallet/chancloser/rbf_coop_states.go

```diff
@@ -241,6 +241,11 @@ type ChanStateObserver interface {
 	// new outgoing add messages.
 	DisableOutgoingAdds() error
 
+	// DisableChannel attempts to disable a channel (marking it ineligible
+	// to foward), and also sends out a network udpate to disable the
+	// channel.
+	DisableChannel() error
```

Comment:
You suggest we split up this interface? 

This ends calling into the `netann` package, to send a disable updated in the gossip network.

Reply:
> I think the essential question I'm getting at is whether or not the chancloser's logic should be tasked with making calls that change the state of the channel

So initially I had many of these as daemon events, so emitted by the state machine as an external event, then executed by the main state machine executor. The feedback was that stuff like disabling a link wasn't universal, so didn't belong in the main executor. In response, I moved that to be a normal emitted state machine. 

> In general, whenever you are putting functions into an object that allow it to make mutations to external systems, you can transform the code such that you have the function return a description of the action you want to perform and then the calling context can perform it for you.

Yep this is exactly how the daemon events work: the state machine emits a description of what they want done (done a message, etc), and then the daemon executes it. I walked back on this partially though based on earlier feedback, see above. 

>  My guess is that in go, the ergonomics of this transform may feel heavy, and maybe the isolation guarantees aren't that much of a win for being able to reason about the code.

One way to do that would be to return to the prior style, and embrace it further, s.t we only use the components of the env to _read_ state. Then we can pass in some custom executor functions (each new state machine can pick which components to initialize), and have all the write/update actions be handled by the executor. 

An advantage of having it in the env as is, is that the unit tests can mock the env interfaces, and then assert that they've been called in the proper order w/ args, etc. If we move them all back into the executor, then we lose that _direct_ assertion, but ofc can still assert that the proper external event was emitted. 

At this point, I prefer to focus on any cross implementation quirks rather than doing additional revisions of the greater framework. In the future, if we continue to use the framework (for say splicing or something), then we'd have an opportunity to rework things to describe any desired mutations via the external event structure.